connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -1532,6 +1532,11 @@ class SparkConnectPlanner(val sessionHolder: SessionHolder) extends Logging {
               s"Failed to load class correctly due to $nsm. " +
                 "Make sure the artifact where the class is defined is installed by calling" +
                 " session.addArtifact.")
+          case cnf: ClassNotFoundException =>
```

Comment:
This is great!

One thing to think about: If there is no serialization proxy involved it should be relatively easy to figure out which class to add. However if a proxy involved you'll get a ClassCastException that is very cryptic. See 
 https://issues.apache.org/jira/browse/SPARK-36917. We could also try to make that a bit more actionable.

Reply:
I couldn't reproduce this in the context of connect. Can you help with getting a repro of this issue in connect, and I'll be happy to look into this?